### PR TITLE
Prototype using sirun for benchmarks

### PR DESCRIPTION
**Pull request by brettlangdon**
**Depends on #2305**

Quick prototype of what adding [sirun](https://github.com/DataDog/sirun) benchmarks to the repo.

Adding in a startup benchmark, and one for flask.

### CODE REVIEW

File: benchmarks/run [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+def main(base_dir, suites):
+    # type: (str, typing.List[str]) -> None
+    if not suites:
+        suites = [os.path.basename(fname) for fname in glob.glob(os.path.join(base_dir, "*")) if os.path.isdir(fname)]
+
+    for suite_name in suites:
+        suite_dir = os.path.join(base_dir, suite_name)
+        results = run_suite(suite_dir)
+        json.dump(results, sys.stdout)
```
Comment (P403n1x87) [inline benchmarks/run:97]: Can (or will) the data be uploaded somewhere, like an S3 bucket?

Comment (brettlangdon) [reply benchmarks/run:97]: yeah, this is missing, forgot to add as a TODO to the description.

I'd like to store as an artifact on the CircleCI job, and also find a way to comment on the results on any GitHub PR.

Comment (brettlangdon) [reply benchmarks/run]: ok, this is updated now to store results as a json file in `benchmarks/results/results.<pid>.json`, and they will be added to CircleCI artifact.

once we review/test this a bit more, I'd like to also add a final job/task that aggregates all the data from all the result json files into something useful, image? tables? etc, and if we can.. post the results in the GitHub PR comments.

File: riotfile.py [hunk 2]
```diff
@@ -138,10 +138,19 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
             command="scripts/build-docs",
         ),
         Venv(
-            name="benchmarks",
-            pys=select_pys(),
-            pkgs={"pytest-benchmark": latest, "msgpack": latest},
-            command="pytest --no-cov {cmdargs} tests/benchmarks",
+            pys=select_pys(min_version="3.6"),  # TODO: This is because rich is only supported on >=3.6
+            pkgs={"rich": latest},
```
Comment (brettlangdon) [inline riotfile.py:142]: `rich` is nice, but might be better to drop it do we can run on python 2.7

**Pull request closed without merging**